From 7e865e838da11b2d485f1aacbc233261e5056f06 Mon Sep 17 00:00:00 2001
From: Paul Jakma <paul@quagga.net>
Date: Mon, 13 Feb 2012 13:53:07 +0000
Subject: [PATCH 3/4] bgpd: Fix regression in args consolidation, total should
 be inited from args

* bgp_attr.c: (bgp_attr_unknown) total should be initialised from the args.
---
 bgpd/bgp_attr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bgpd/bgp_attr.c b/bgpd/bgp_attr.c
index 0d82aba..81802ce 100644
--- a/bgpd/bgp_attr.c
+++ b/bgpd/bgp_attr.c
@@ -1646,7 +1646,7 @@ bgp_attr_ext_communities (struct bgp_attr_parser_args *args)
 static bgp_attr_parse_ret_t
 bgp_attr_unknown (struct bgp_attr_parser_args *args)
 {
-  bgp_size_t total;
+  bgp_size_t total = args->total;
   struct transit *transit;
   struct attr_extra *attre;
   struct peer *const peer = args->peer; 
-- 
1.8.0

